File: AbetOrder/OpenDataAccess/OpenDataAccess.Entity/IValidate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenDataAccess.Entity
{
    public interface IValidate
    {
        string InsertValidateUnique<T>(IEntityData entityData, string propertyName, string message) where T : IEntity;
        string UpdateValidateUnique<T>(IEntityData entityData, string propertyName, string message) where T : IEntity;
        string ValidateExists<T>(IEntityData entityData, string filter, string message, bool blExists = true) where T : IEntity;
        string Validate();
        string Validate(IEntityData entityData, List<Func<IValidate, IEntityData, string>> validateList);
    }
}

File: JavaOpenDataAccess/src/OpenDataAccess/Entity/IValidate.java
package OpenDataAccess.Entity;

import OpenDataAccess.LambdaInterface.IFunction2;

import java.util.List;

public interface IValidate {
    <T extends IEntity> String InsertValidateUnique(IEntityData entityData, String propertyName, String message);
    <T extends IEntity> String UpdateValidateUnique(IEntityData entityData, String propertyName, String message);
    <T extends IEntity> String ValidateExists(IEntityData entityData, String filter, String message, boolean blExists);
    String Validate();
    String Validate(IEntityData entityData, List<IFunction2<IValidate, IEntityData, String>> validateList);
}
